sql-ballerina/tests/complex-query-test.bal

```diff
@@ -691,3 +691,34 @@ function testGetArrayTypes4() returns error? {
     };
     test:assertEquals(value, expectedData, "Expected data did not match.");
 }
+
+type UserDefinedTypes record {
+    int udt_int;
```

Comment:
You have to use typed values in order to hit UDT code,

Check out https://github.com/ballerina-platform/module-ballerinax-postgresql/blob/6286b237e29d32741a51a8f0f985e105ab1e406b/postgresql-ballerina/tests/execute-params-test.bal#L1579

Reply:
Yes, But in order to support that need to do some code changes. it seems that currently we do not support custom SQL types in SQL module. https://github.com/ballerina-platform/module-ballerina-sql/blob/e9910ebaaa9fc29fcee79f7c2a06146353dd194b/sql-native/src/main/java/org/ballerinalang/sql/parameterprocessor/DefaultStatementParameterProcessor.java#L1146

Reply:
Shall we create a negative test case, we ask for a custom type even though the DB is not supported. This way we can hit these lines.

Reply:
-1, we can not create negative test cases without creating custom types in DB. because in SQL module we throw ApplicationError when there is a custom type is requested.

Reply:
Yes, I am saying is to send the query and validate that this is throwing an error.

Reply:
removed this test case in https://github.com/ballerina-platform/module-ballerina-sql/pull/291/commits/a24fa36382ca4de8797472c84247e76d65a4734e